### PR TITLE
refactor(command): migrate write_tree to Git::Commands::WriteTree

### PR DESCRIPTION
**Pull request by jcouball**
## Summary

Migrates `Git::Lib#write_tree` to a new `Git::Commands::WriteTree` class as part of the Phase 2 Strangler Fig pattern migration (checklist item 48/54).

### New: `lib/git/commands/write_tree.rb`

`Git::Commands::WriteTree` wraps `git write-tree` using the `Git::Commands::Base` architecture. Exposes all options documented in the minimum-supported git version (2.28.0):

- `--missing-ok` — disable check that all referenced objects exist in the object database
- `--prefix=<prefix>/` — write a tree for a subdirectory instead of the full index

### Updated: `lib/git/lib.rb`

`Git::Lib#write_tree` now delegates to `Git::Commands::WriteTree`:

```ruby
def write_tree
  Git::Commands::WriteTree.new(self).call.stdout
end
```

### Tests

- **Unit tests** (`spec/unit/git/commands/write_tree_spec.rb`): 5 examples covering no-options baseline, each option individually, combined options, and unsupported-option validation.
- **Integration tests** (`spec/integration/git/commands/write_tree_spec.rb`): 3 examples verifying real git behavior — SHA format, object type, and prefix subdirectory tree creation.

### Docs

- `redesign/3_architecture_implementation.md` — marked `write_tree` done, updated progress counter to 48/54, updated Next Task to `gc`.

## Checklist

- [x] Unit tests pass (`bundle exec rspec spec/unit/git/commands/write_tree_spec.rb`)
- [x] Integration tests pass (`bundle exec rspec spec/integration/git/commands/write_tree_spec.rb`)
- [x] Full RSpec suite passes (3362 examples, 0 failures)
- [x] Legacy TestUnit suite passes (557 tests, 0 failures)
- [x] Rubocop clean
- [x] Implementation plan updated

### CODE REVIEW

File: spec/integration/git/commands/write_tree_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    repo.add('hello.txt')
+  end
+
+  describe '#call' do
+    context 'when the command succeeds' do
+      it 'returns a CommandLineResult with the tree SHA on stdout' do
+        result = command.call
+
+        expect(result).to be_a(Git::CommandLineResult)
+        expect(result.stdout).to match(/\A[0-9a-f]{40}\z/)
```
[the remark this answers is not in the document]
Comment (jcouball) [reply spec/integration/git/commands/write_tree_spec.rb:23]: Not doing this.

Looking at the redesign doc's guideline:

"Only add essential integration tests for edge cases and testing that the assumptions for git output used in unit tests are correct."

The feedback is essentially asking for tests that prove git itself is correct.
